src/app/tracing/io/data-importer/xlsx-import/shared.ts

```diff
@@ -0,0 +1,616 @@
+                columnMappings.push({
+                    fromIndex: columnIndex,
+                    toPropId: colRef,
+                    type: mappingDef[colRef]!.type,
+                });
+            }
+        }
+    });
+    return columnMappings;
+}
+
+export function getCleanedStringOrUndefined(
+    value: CellValue | undefined,
+): string | undefined {
+    if (typeof value === "string") {
+        value = value.trim();
+        if (value === "") {
+            value = undefined;
+        }
+    } else if (value !== undefined) {
+        return `${value}`;
+    }
+    return value;
+}
+
+function getCleanedInput(value: CellValue | undefined): CellValue | undefined {
+    if (typeof value === "string") {
+        value = value.trim();
+        if (value === "") {
+            value = undefined;
+        }
+    }
+    return value;
+}
+
+function getLat(lat: any): number | undefined {
+    return typeof lat === "number" &&
+        !Number.isNaN(lat) &&
+        isInRange(lat, LATITUDE_LIMITS.min, LATITUDE_LIMITS.max)
+        ? lat
+        : undefined;
+}
+
+function getLon(lon: any): number | undefined {
+    return typeof lon === "number" &&
+        !Number.isNaN(lon) &&
+        isInRange(lon, LONGITUDE_LIMITS.min, LONGITUDE_LIMITS.max)
+        ? lon
+        : undefined;
+}
+
+function getBoolean(value: any): boolean | undefined {
+    return typeof value === "boolean" ? value : undefined;
+}
+
+function getNumber(value: any): number | undefined {
+    return typeof value === "number" && !Number.isNaN(value)
+        ? value
+        : undefined;
+}
+
+export function getStringOrUndefined(value: any): string | undefined {
+    return typeof value === "string" ? value : undefined;
+}
+
+function getNonNegNumber(value: any): number | undefined {
+    return typeof value === "number" && !Number.isNaN(value) && value >= 0
+        ? value
+        : undefined;
+}
+
+const TYPESTRING_2_FUN = {
+    lat: getLat,
+    lon: getLon,
+    "nonneg:number": getNonNegNumber,
+    number: getNumber,
+    string: getStringOrUndefined,
+    boolean: getBoolean,
+} as const satisfies {
+    [T in RefinedTypeString]: (
+        x: CellValue | undefined,
+    ) => TypeString2Type<T> | undefined;
+};
+
+function conditionalConcat(
+    arr: (string | number | boolean | undefined)[],
+    sep: string,
+): string | undefined {
+    arr = removeUndefined(arr);
+    const filteredArr = removeUndefined(arr);
+    return filteredArr.length === 0 ? undefined : filteredArr.join(sep);
+}
+
+export function importValue<X extends RefinedTypeString>(
+    row: Row,
+    table: Table,
+    colIndex: number,
+    reqType: X,
+    addIssueCb: AddIssueCallback,
+    required = false,
+    invalidateRow = false,
+): TypeString2Type<X> | undefined {
+    const inputValue = getCleanedInput(row[colIndex]);
+    let value: TypeString2Type<X> | undefined;
+    if (inputValue === undefined) {
+        if (required === true) {
+            addIssueCb(
```

Comment:
same here: if there is a way to simplify the nested conditionals, I'd recommend that we do that

Reply:
I do not know how to unnest this without double checking things.